archive-cleaner

```diff
@@ -68,32 +67,48 @@ def get_packages_from_buildinfo(archive, filename, archive_entry, arch):
                 continue
-            directory = join(archive, 'packages', pkgname[0], pkgname)
-            path = join(directory, pkgfilename)
-            if not isfile(path):
-                print(f'ERROR: {package} needed for {filename} not in archive {path}', file=stderr)
+            directory = join(archive, 'packages'. pkgname[0], pkgname)
+            paths = [(pkgfilename, join(directory, pkgfilename)) for pkgfilename in pkgfilenames]
+            existing_paths = list(filter(lambda path_tuple: isfile(path_tuple[1]), paths))
+            if len(existing_paths) == 0:
+                for _, path in paths:
+                    print(f'ERROR: {package} needed for {filename} not in archive {path}', file=stderr)
                 continue
-            packages.append(pkgfilename)
+            packages.extend(list(map(lambda path_tuple: path_tuple[0], existing_paths)))
         return packages
     for package in installed:
-        filename_any = f'{package}-any.pkg.tar.xz'
-        filename_arch = f'{package}-{arch}.pkg.tar.xz'
+        filenames_any = [f'{package}-any.pkg{ext}' for ext in PKG_EXTENSIONS]
+        filenames_arch = [f'{package}-{arch}.pkg{ext}' for ext in PKG_EXTENSIONS]
         pkgname = '-'.join(package.split('-')[:-2])
         directory = join(archive, 'packages', pkgname[0], pkgname)
-        path_any = join(directory, filename_any)
-        path_arch = join(directory, filename_arch)
-        if isfile(path_any):
-            packages.append(filename_any)
-        elif isfile(path_arch):
-            packages.append(filename_arch)
-        else:
-            print(f'ERROR: {package} needed for {filename} not in archive {path_any}', file=stderr)
-            print(f'ERROR: {package} needed for {filename} not in archive {path_arch}', file=stderr)
+        paths_any= [(filename_any, join(directory, filename_any)) for filename_any in filenames_any]
+        paths_arch = [(filename_arch, join(directory, filename_arch)) for filename_arch in filenames_arch]
+
+        found_package = None
+        for path_any_tuple, path_arch_tuple in zip(paths_any, paths_arch):
+            filename_any, path_any = path_any_tuple
+            filename_arch, path_arch = path_arch_tuple
+            if isfile(path_any):
+                found_package = filename_any
+                break
+            elif isfile(path_arch):
+                found_package = filename_arch
+                break
+
+        if found_package is None:
+            for path_any_tuple, path_arch_tuple in zip(paths_any, paths_arch):
+                filename_any, path_any = path_any_tuple
+                filename_arch, path_arch = path_arch_tuple
+                print(f'ERROR: {package} needed for {filename} not in archive {path_any}', file=stderr)
+                print(f'ERROR: {package} needed for {filename} not in archive {path_arch}', file=stderr)
+                continue
+        packages.append(found_package)
+
```

Comment:
This looks way too complex than it should be, also i don't understand why we need to pass around tuples again.
I would like to see this whole block refactored in a much simpler approach, like the one before, just with `map`-ing `isfile` function to the array of possibilities.
Especially it doesn't make sense anymore to print error for all possible combination of pkgext, just check a single result array is empty (which a `not arrayvar` check instead of using `len`) and output a proper error description once for all ext combinations.

Reply:
Okay will use filter and map again, I just printed all errors because I didn't want to change the error message